File: eyan_helper/src/main/scala/eu/eyan/util/time/TimeCounter.scala
package eu.eyan.util.time

import eu.eyan.log.Log

object TimeCounter {
  def now = System.currentTimeMillis
  def countAndPrint(msg: String)(action: => Unit) = Log.info(s"$msg ${millisecsOf(action)} ms") 

  def millisecsOf(action: => Unit) = {
    val start = now
    action
    now - start
  }
}

File: src/main/java/eu/eyan/timecounter/TimeCounter.java
package eu.eyan.timecounter;


public class TimeCounter {
	private long myStart;

	public void start(){
		this.myStart = System.currentTimeMillis();
	}
	
	public String end(){
		long ms = System.currentTimeMillis()-myStart;
		float seconds = ((float)(ms/1000));
		return ""+seconds+"s";
	}
	
}
